src/HeaderBar/Apps.js

```diff
@@ -0,0 +1,288 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { Card, InputField, colors, theme } from '@dhis2/ui-core'
+
+import { gotoURL } from '../utils/url.js'
+
+import { Settings } from '../icons/Settings.js'
+import { Apps as AppsIcon } from '../icons/Apps.js'
+import { Cancel } from '../icons/Cancel.js'
+
+import cx from 'classnames'
+
+import css from 'styled-jsx/css'
+
+import { DataRequest } from '@dhis2/app-service-data'
+
+const appIcon = css.resolve`
+    svg {
+		fill: ${colors.white};
+        cursor: pointer;
+		height: 24px;
+		width: 24px;
+    }
+`
+
+const trailIcon = css.resolve`
+    svg {
+		fill: ${colors.grey900};
+        cursor: pointer;
+		height: 24px;
+		width: 24px;
+		margin-right: 8px;
+		margin-top: 4px;
+    }
+`
+
+const settingsIcon = css.resolve`
+    svg {
+        margin: 8px 8px 0 16px;
+        color: ${colors.grey900};
+		height: 24px;
+		width: 24px;
+        cursor: pointer;
+    }
+`
+
+function TrailIcon({ onClick }) {
+    return (
+        <a onClick={onClick}>
+            <Cancel className={trailIcon.className} />
+        </a>
+    )
+}
+
+function Search({ value, onChange, onSettingsClick, onIconClick }) {
```

Comment:
Could be moved to `ui-core` to have a generalized `Search` input field.

Reply:
> I guess all then `src/icons/[name].js` components are used in the Headerbar only.
> If so, I think we can move the `icons` folder into the `HeaderBar` folder,
> otherwise, I'd move them to `ui-core` as they're dumb components rather than widgets.

535b647

Reply:
> Could be moved to `ui-core` to have a generalized `Search` input field.

The "settings" part of this component makes it special to the headerbar. It's the only place it's used right now in any UX.